Tweak: Added new units for the social icons widget and the icon list …

rodolphebertozzo opened the pull request.

rodolphebertozzo commented:
@arielk

rodolphebertozzo commented:
@KingYes

rami-elementor commented:
I think we need to add default `'unit' => ...` to all the places where new units where introduced.

In addition, you should add a `range` argument setting min/max/step for each unit, where it required.

rodolphebertozzo commented:
Hi @rami-elementor, I added default units everywhere, I put the px logically
As for the steps, I added them everywhere (0.1) and concerning the minimum and maximum values, I added them respecting a logic, where max was 1000 for the px, I logically put 100 for the %, when the values were lower, I added maximum values in a proportional way, for example : 100px = 10%.

rodolphebertozzo commented:
@hein-obox, can you review please

arielk commented on includes/widgets/social-icons.php:
```diff
@@ -427,14 +446,30 @@ protected function register_controls() {
+				'size_units' => [ 'px', '%', 'em', 'rem' ],
+				'default' => [
+					'unit' => 'px',
+					'size' => 14,
```
@rodolphebertozzo please change the size default to `'size' => 5,`

rodolphebertozzo replied:
Done @arielk !

arielk commented on includes/widgets/icon-list.php:
```diff
@@ -296,14 +296,24 @@ protected function register_controls() {
 					],
+					'em' => [
+						'max' => 5,
+						'step' => 0.1,
```
@rodolphebertozzo The default of the `em` and `rem` is:
```
'em' => [
		'min' => 0.1,
		'max' => 10,
		'step' => 0.1,
	],
'rem' => [
		'min' => 0.1,
		'max' => 10,
		'step' => 0.1,
	],
```
Learn more here: https://github.com/elementor/elementor/blob/master/includes/controls/base-units.php#L46

So you don't need to add this again in the control it self, only if you want to change the default values.

rodolphebertozzo replied:
Done.

arielk commented on includes/widgets/icon-list.php:
```diff
@@ -320,9 +330,22 @@ protected function register_controls() {
+					'px' => [
+						'min' => 1,
+						'max' => 1000,
+						'step' => 0.1,
```
remove the range

rodolphebertozzo replied:
Done.

arielk commented on includes/widgets/icon-list.php:
```diff
@@ -424,12 +447,29 @@ protected function register_controls() {
+					'unit' => 'px',
 					'size' => 14,
 				],
 				'range' => [
```
Change to:
```
'range' => [
		'px' => [
			'min' => 6,
			],
		'%' => [
			'min' => 5,
			],
		'em' => [
			'min' => 1,
			],
		'rem' => [
			'min' => 1,
			],
	],
```

The reason why we added the min value to avoid situation that the icon is hidden when the user choose `0` value

rodolphebertozzo replied:
Done.

arielk commented on includes/widgets/image-gallery.php:
```diff
@@ -207,14 +207,30 @@ protected function register_controls() {
+				],
 				'range' => [
 					'px' => [
+						'min' => 0,
```
remove `min` and `step` for `px`

rodolphebertozzo replied:
Done.

arielk commented on includes/widgets/image-gallery.php:
```diff
@@ -207,14 +207,30 @@ protected function register_controls() {
 						'max' => 100,
+						'step' => 0.1,
+					],
+					'%' => [
```
Remove range for % em and rem

rodolphebertozzo replied:
Done.

arielk commented on includes/widgets/image-gallery.php:
```diff
@@ -207,14 +207,30 @@ protected function register_controls() {
+				'size_units' => [ 'px', '%', 'em', 'rem' ],
 				'show_label' => false,
+				'default' => [
+					'unit' => 'px',
```
Add 'size' => 15, to default

rodolphebertozzo replied:
Done.

arielk commented on includes/widgets/social-icons.php:
```diff
@@ -384,10 +384,29 @@ protected function register_controls() {
 					'px' => [
 						'min' => 6,
 						'max' => 300,
+						'step' => 0.1,
```
See my comment about icon-list

rodolphebertozzo replied:
Done.

arielk commented on includes/widgets/social-icons.php:
```diff
@@ -447,9 +482,29 @@ protected function register_controls() {
+					'unit' => 'px',
 					'size' => 0,
 				],
+				'range' => [
```
Same note as above

rodolphebertozzo replied:
Done.

rami-elementor commented on includes/widgets/social-icons.php:
```diff
@@ -384,10 +384,23 @@ protected function register_controls() {
 				'label' => esc_html__( 'Size', 'elementor' ),
 				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', '%', 'em', 'rem' ],
+				'default' => [
+					'unit' => 'px',
+					'size' => 14,
+				],
```
New defaults values?

rami-elementor replied:
@rodolphebertozzo This is probably the reason why the tests fail.

rodolphebertozzo replied:
Done @rami-elementor

rami-elementor commented:
Generally speaking, I like this PR because I have a task to standardize units in various icon controls.

rodolphebertozzo committed:
Delete package-lock.json

rodolphebertozzo committed:
Merge branch 'release/3.8.0' into patch-12

rodolphebertozzo committed:
Merge branch 'release/3.8.0' into patch-12
